src/domains/collections.ts

```diff
@@ -186,105 +213,116 @@ export const collectionToStac = (collection: any): STACCollection => {
-  facets: FacetGroup | null;
   cursor: string | null;
   items: STACCollection[];
 }> => {
-  const { headers = {} } = opts;
-
-  const clientId = buildClientId(headers["client-id"]);
-  const authorization = headers.authorization;
-
-  const requestHeaders = mergeMaybe(
-    { "client-id": clientId },
-    { authorization }
-  );
+  return await paginateQuery(collectionsQuery, params, opts, collectionHandler);
+};
 
-  const timingMessage = `Outbound GQL collections query => ${JSON.stringify(
-    query,
-    null,
-    2
-  )} ${JSON.stringify(scrubTokens(requestHeaders), null, 2)}`;
-  console.time(timingMessage);
-  const {
-    collections: { count, cursor, items, facets },
-  } = await request({
-    url: GRAPHQL_URL,
-    document: collectionsQuery,
-    variables: { params: query },
-    requestHeaders,
-  });
-  console.timeEnd(timingMessage);
-
-  return { count, cursor, facets, items: items.map(collectionToStac) };
+const collectionIdsHandler: GraphQLHandler = (response: any) => {
+  try {
+    const {
+      collections: { count, items, cursor },
+    } = response;
+
+    return [
+      null,
+      {
+        count,
+        cursor,
+        items,
+      },
+    ];
+  } catch (err) {
+    return [(err as Error).message, null];
+  }
 };
 
 /**
  * Fetches only collectionIds.
  */
 export const getCollectionIds = async (
-  query: CollectionsInput,
+  params: CollectionsInput,
   opts: {
     headers?: { "client-id"?: string; [key: string]: any };
     [key: string]: any;
   } = {}
 ): Promise<{
   count: number;
   cursor: string | null;
-  conceptIds: string[];
+  items: { conceptId: string; title: string }[];
 }> => {
-  let userClientId = "cmr-stac";
-  let authorization;
-
-  const { headers } = opts;
-  if (headers) {
-    userClientId = buildClientId(headers["client-id"]);
-    authorization = headers.authorization;
-  }
-
-  const requestHeaders = mergeMaybe(
-    { "client-id": userClientId },
-    { authorization }
+  return await paginateQuery(
+    collectionIdsQuery,
+    params,
+    opts,
+    collectionIdsHandler
   );
+};
 
-  const timingMessage = `Outbound GQL collectionIds query => ${JSON.stringify(
-    query,
-    null,
-    2
-  )} ${JSON.stringify(scrubTokens(requestHeaders), null, 2)}`;
-  console.time(timingMessage);
-  const {
-    collections: { count, cursor, items },
-  } = await request({
-    url: GRAPHQL_URL,
-    document: collectionIdsQuery,
-    variables: { params: query },
-    requestHeaders,
-  });
-  console.timeEnd(timingMessage);
-
-  const conceptIds = items.map((coll: { conceptId: string }) => coll.conceptId);
-
-  return { count, cursor, conceptIds };
+export const getAllCollectionIds = async (
+  params: CollectionsInput,
+  opts: {
+    headers?: { "client-id"?: string; [key: string]: any };
+    [key: string]: any;
+  } = {}
+): Promise<{
+  count: number;
+  cursor: string | null;
+  items: { conceptId: string; title: string }[];
+}> => {
+  params.limit = MAX_SIGNED_INTEGER;
```

Comment:
I may be incorrect but, this is the limit field is defaulting really to 1000 right since that is the maximum number of concepts that CMR will return per page?

Reply:
This is once it's past the user validation so not subject to the same limitation.